File: clojure/src/git_remote_s3/enums.clj
; SPDX-FileCopyrightText: Amazon.com, Inc. or its affiliates
;
; SPDX-License-Identifier: Apache-2.0

(ns git-remote-s3.enums
  "URI scheme constants, mirroring the Python UriScheme enum.")

;; The two supported URI schemes are represented as Clojure keywords.
(def uri-scheme-s3    :s3)
(def uri-scheme-s3zip :s3+zip)

;; Set used for membership testing / validation.
(def all-uri-schemes #{uri-scheme-s3 uri-scheme-s3zip})

(defn valid-uri-scheme?
  "Returns true when `scheme` is one of the known URI schemes."
  [scheme]
  (contains? all-uri-schemes scheme))

File: Sources/GitRemoteS3/Enums.swift
// SPDX-FileCopyrightText: Amazon.com, Inc. or its affiliates
// SPDX-License-Identifier: Apache-2.0
//
// Translated from git_remote_s3/enums.py

/// URI schemes supported by git-remote-s3.
public enum UriScheme: String, Equatable {
    case s3    = "s3"
    case s3Zip = "s3+zip"
}
